src/Psl/Iter/search_opt.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Iter;
+
+use Closure;
+use Psl\Option\Option;
+
+/**
+ * Searches an iterable until a predicate returns true, then returns
+ * the value of the matching element wrapped in {@see Option::some}.
+ * If a predicate never returns true, {@see Option::none} will be returned.
+ *
+ * Examples:
+ *
+ *      Iter\search(['foo', 'bar', 'baz'], fn($v) => 'baz' === $v)
+ *      => Str('baz')
+ *
+ *      Iter\search(['foo', 'bar', 'baz'], fn($v) => 'qux' === $v)
+ *      => Null
```

Comment:
exampless are outdated

Reply:
good catch. is it correct now? Not sure what `Str('baz')` was, replaced.